src/store/pageview.ts

```diff
@@ -17,6 +26,29 @@ class ViewStore {
     storage.set(StorageKeys.PAGE_VIEWS, this.pageViews);
   }
 
+  _trim() {
+    this.pageViews.sort((a: PageView, b: PageView): number => b.ts - a.ts);
```

Comment:
The page views are already inserted by time, so the last one is the most recent.

Also you are calling sort even if we do not need to clean anything up.

Reply:
Here we have two concurrent questions:
- The algorithm correctness;
- The function being pure;

So:
- If you consider just the happy path, where you rely on a particular state (the localStorage being filled in order and not being touched), then 'no sorting' makes your function correct but not pure;
- If you want your function to be pure and easier to reason about for those who does not know the state affairs, then you cannot rely on the state. This way it is also correct for any input state.

Also, sorting is supposed to be fast `O(n log n)`. I would opt for correctness and pureness over time savings, at least if speed is not a really critical issue.

WDYT?

Reply:
Ahhh I knew the functional stuff would come up :)

In all honestly I agree and I think that adding a line of code to make sure it is sorted is a good idea.

The issue is, all of our business logic is filter, sort, maps etc the more features we add the more this becomes an issue - our SDK is essentially blocking for most of the logic, and for the user this can be an annoyance, as the page will be frozen.

So our optimisation goal is not: "make the fastest sdk" it is "do not annoy the user".

Even if we put this aside, we are doing sort -> filter on all elements each time this is called, we could optimise:
- reverse the iteration for max age
- cut off the over max size elements first
- (insert better ideas)....

Reply:
I agree that 'not annoying our users' is a good business decision! haha

Reply:
Do you agree that we do not need to sort if the length is below the max?

Check length, if over max, then sort, then slice?